hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -402,6 +402,8 @@ public CustomPayloadBufferedRecordMerger(
         Schema readerSchema,
         TypedProperties props) {
       super(readerContext, recordMerger, readerSchema, props);
+      ValidationUtils.checkArgument(!Objects.equals(payloadClass, OverwriteWithLatestAvroPayload.class.getCanonicalName())
```

Comment:
why we need this? The merger should not care about specific impl of the payload IMO, the check should be done before the merger is created.

Reply:
I think we can, as long as the merge mode inference works. Otherwise it should be a bug.